Allow 'telemetry.checks' to be set through env vars

hush-hush opened the pull request:
### What does this PR do?

Allow `telemetry.checks` to be set through env vars.

This also ignore `config.BindEnv` error not being checked since it was ignore everywhere anyway.

### Checklist
<!-- Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields. -->

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

hush-hush committed:
Allow 'telemetry.checks' to be set through env vars

olivielpeau commented on pkg/config/config.go:
```diff
@@ -761,7 +761,7 @@ func InitConfig(config Config) {
 	// This create a lot of billable custom metrics.
 	config.BindEnvAndSetDefault("telemetry.enabled", false)
-	config.SetKnown("telemetry.checks")
+	config.BindEnvd("telemetry.checks")
```
```suggestion
	config.BindEnv("telemetry.checks")
```

hush-hush committed:
Remove the error returned by config.BindEnv

We never check the error and it's counter part BindEnvAndSetDefault
doesn't return an error either.

olivielpeau commented on .golangci.yml:
```diff
@@ -25,6 +25,7 @@ issues:
     - "`defaultRetryDuration` is unused" # used by APM and Process
     - "`defaultRetries` is unused"       # used by APM and Process
     - "python._Ctype_char, which can be annoying to use" # ignore warning about returning unexported field from CGO
+    - "return value of `config.BindEnv` is not checked" # ignore BindEnv error return not being checked
```
I agree that the error can be ignored (only case when it can return an error is https://github.com/DataDog/viper/blob/8e7459bcf337c155af4be4d26f6d636f1204a1fb/viper.go#L965, which we can completely ignore).

That said, instead of this, I think we should modify the `Config` interface and `pkg/config/viper.go` to simply make `BindEnv` not return any error. WDYT?

mx-psi replied:
➕ I was going to comment the same thing

hush-hush replied:
I thought of this, but the official viper interface returns an error https://pkg.go.dev/github.com/DataDog/viper#BindEnv and I think we want to stay similar to it.

olivielpeau replied:
I don't think we're aiming for the `Config` interface to match the `viper` lib exactly to be honest. Although I agree that it's better for our viper fork to have the same function signatures as upstream.

brycekahle commented on pkg/config/system_probe.go:
```diff
@@ -76,7 +76,7 @@ func InitSystemProbeConfig(cfg Config) {
 	// we cannot use BindEnvAndSetDefault for network_config.enabled because we need to know if it was manually set.
 	cfg.SetKnown(join(netNS, "enabled"))
-	_ = cfg.BindEnv(join(netNS, "enabled"), "DD_SYSTEM_PROBE_NETWORK_ENABLED")
+	cfg.BindEnv(join(netNS, "enabled"), "DD_SYSTEM_PROBE_NETWORK_ENABLED") //nolint:errcheck
```
Any reason to switch to the magic comment? I think the explicit error ignore with `_` is more obvious, and doesn't rely upon a linting tool supporting a specific magic comment.
